### PR TITLE
gh: run unit tests on Ubuntu and Windows

### PR DESCRIPTION
**Pull request by mmatczuk**
Fix Martian connection error handling on Windows and run unit tests on Windows runner.

### CODE REVIEW

File: runctx/runctx_unix_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 Sauce Labs Inc. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//go:build unix
```
Comment (Choraden) [inline runctx/runctx_unix_test.go:7]: suggestion: `go:build unix` or `go:build !windows`

File: internal/martian/proxy.go [hunk 2]
```diff
@@ -89,18 +89,23 @@ func isClosedConnError(err error) bool {
...
 
-	if errors.Is(err, io.EOF) ||
-		errors.Is(err, io.ErrClosedPipe) ||
-		errors.Is(err, errClose) {
+	if strings.Contains(err.Error(), "tls:") {
 		return true
 	}
 
-	return strings.Contains(err.Error(), "tls:")
+	return false
```
Comment (Choraden) [inline internal/martian/proxy.go:108]: Why so passive aggressive? :)

Let's just return `strings.Contains(err.Error(), "tls:")`

File: runctx/runctx_unix_test.go (hunk 1 above)
Comment (mmatczuk) [reply runctx/runctx_unix_test.go:7]: I do not get it? It is `//go:build unix`, `go:build !windows` does not apply here.

File: .golangci.yml [hunk 3]
```diff
@@ -78,6 +78,7 @@ issues:
     - "Error return value of `cmd.MarkFlag.+` is not checked"
     - "Error return value of `net.SplitHostPort` is not checked"
     - "Error return value of `w.Write` is not checked"
+    - "Function 'handle' has too many statements"
```
Comment (Choraden) [inline .golangci.yml:81]: We keep silence these linters or increase limits. Maybe we should disable it for good.

File: runctx/runctx_unix_test.go (hunk 1 above)
Comment (Choraden) [reply runctx/runctx_unix_test.go:7]: I meant to use this implementation for all platforms that are not windows, not only unix.

Comment (Choraden) [reply runctx/runctx_unix_test.go:7]: But maybe it's better to be straightforward here.

File: config_test.go [hunk 4]
```diff
@@ -252,6 +253,10 @@ func TestParseFilePath(t *testing.T) {
 
 	for i := range tests {
 		tc := &tests[i]
+		if runtime.GOOS == "windows" {
+			tc.err = ""
+		}
```
[the remark this answers is not in the document]
Comment (mmatczuk) [reply config_test.go:258]: Added a comment.

**Review by Choraden — APPROVED** [review]

**Comment by mmatczuk** [comment]
> I can see plenty of gh actions were run :)

Yes, as I tried to run all the tests like we do in other places.
To do that I'd need to get a local Windows access first.

File: .golangci.yml (hunk 3 above)
Comment (mmatczuk) [reply .golangci.yml:81]: OK.

**Pull request merged**